local-tests/rolling_update.ipynb

```diff
@@ -0,0 +1,363 @@
+    "sys.path.append(join(alephhome, 'local-tests'))\n",
+    "from chainrunner import Chain, Seq, generate_keys"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8cfb5287-6026-432e-a880-3121cd102547",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "testhome = '/home/hansu/aleph/desk'\n",
+    "# Path to working directory, where chainspec, logs and nodes' dbs are written:\n",
+    "workdir = join(testhome, 'workdir')\n",
+    "# Path to the pre-update aleph-node binary:\n",
+    "binary = join(testhome, 'bin', 'aleph-node-10.1')\n",
+    "# Path to the post-update aleph-node binary:\n",
+    "newbin = join(testhome, 'bin', 'aleph-node-11.1')\n",
+    "# Path to the post-update compiled runtime:\n",
+    "runtime = join(testhome, 'bin','aleph_runtime.64')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a0dd1e2a-5c5f-483b-97e9-142e2447107e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "N_VALIDATORS = 7 #at most 10, unless you extend names list below\n",
+    "# Ports for node 0, consecutive numbers are used for other nodes \n",
+    "PORT = 30334\n",
+    "WS_PORT = 9943\n",
+    "RPC_PORT = 9933\n",
+    "VAL_PORT = 30343"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8b14abd3-19d0-4203-848e-e5222f6e9cc0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "names = ['//Alice','//Bob','//Charlie','//Dave','//Eve','//Ferdie','//George','//Hans','//Iris','//James']\n",
+    "phrases = names[:N_VALIDATORS]\n",
+    "keys = generate_keys(binary, phrases)\n",
+    "\n",
+    "sudo = keys['//Alice']\n",
+    "faucet = keys['//Bob']\n",
+    "val_addrs = ['127.0.0.1:'+str(i) for i in range(VAL_PORT, VAL_PORT+N_VALIDATORS)]\n",
+    "\n",
+    "shutil.rmtree(workdir, True)\n",
+    "chain = Chain(workdir)\n",
+    "chain.bootstrap(binary, keys.values(),\n",
+    "                chain_type='live',\n",
+    "                faucet_account_id=faucet,\n",
+    "                sudo_account_id=sudo,\n",
+    "                raw=True)\n",
+    "\n",
+    "chain.set_flags('validator',\n",
+    "                'unsafe-ws-external',\n",
+    "                'unsafe-rpc-external',\n",
+    "                'no-mdns',\n",
+    "                port=Seq(PORT),\n",
+    "                ws_port=Seq(WS_PORT),\n",
+    "                rpc_port=Seq(RPC_PORT),\n",
+    "                validator_port=Seq(VAL_PORT),\n",
+    "                public_validator_addresses=val_addrs,\n",
+    "                unit_creation_delay=500,\n",
+    "                execution='Native',\n",
+    "                rpc_cors='all',\n",
+    "                rpc_methods='Unsafe',\n",
+    "                state_pruning='archive'\n",
+    "               )\n",
+    "\n",
+    "addresses = [n.address() for n in chain]\n",
+    "chain.set_flags(bootnodes=''.join(addresses), public_addr=addresses)"
```

Comment:
Could you instead run an additional nonvalidator node and use that as the bootnode? This is even closer to our actual setup.

Reply:
As the only bootnode, or just one of many?

Reply:
I would go with only, as this is closer to what is going on irl. Unless you want to restart/update it as part of the test, then add a second one imo.

Reply:
Done. Both ways are possible